sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySpanProcessor.java

```diff
@@ -1,19 +1,95 @@
 package io.sentry.opentelemetry;
 
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
 import io.opentelemetry.context.Context;
 import io.opentelemetry.sdk.trace.ReadWriteSpan;
 import io.opentelemetry.sdk.trace.ReadableSpan;
 import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import io.sentry.DateUtils;
+import io.sentry.DsnUtil;
+import io.sentry.ISpan;
+import io.sentry.ITransaction;
+import io.sentry.Instrumenter;
 import io.sentry.Sentry;
+import io.sentry.SpanId;
+import io.sentry.SpanStatus;
+import io.sentry.TransactionContext;
+import io.sentry.TransactionOptions;
+import io.sentry.protocol.SentryId;
+import io.sentry.protocol.TransactionNameSource;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
-@SuppressWarnings("CatchAndPrintStackTrace")
 public final class SentrySpanProcessor implements SpanProcessor {
 
+  private final @NotNull Map<String, ISpan> spans = new ConcurrentHashMap<>();
+  private final @NotNull List<SpanKind> spanKindsConsideredForSentryRequests =
+      Arrays.asList(SpanKind.CLIENT, SpanKind.INTERNAL);
+  private final @NotNull SpanDescriptionExtractor spanDescriptionExtractor =
```

Comment:
m: Why a new class here? Can't this just be methods on the `SentrySpanProcessor`? I'd rather not extract this out if it's only going to be used once.

Reply:
I was expecting it to grow as we add support for more and more types. I extracted it to make `SpanProcessor` smaller as it seemed to be nicely isolated. If we decide to no longer do any extraction on the SDKs we can just delete it.

Reply:
Fair, we can just keep it then.

We'll probably move all the logic server-side eventually, so I guess co-locating the client-side transforms is a good idea for now.